tests/gateway/test_client.py

```diff
@@ -471,3 +471,26 @@ def test_cllient_query_with_disallowed_param(mixed_gateway):
 
     with pytest.raises(HTTPError, match=".*The parameter 'model' is not permitted.*"):
         gateway_client.query(route=route.name, data=data)
+
+
+def test_query_timeout_not_retried(mixed_gateway):
+    gateway_client = MlflowGatewayClient(gateway_uri=mixed_gateway.url)
+
+    data = {"prompt": "Test", "temperature": 0.4}
+    route = "completions"
+
+    with mock.patch(
+        "mlflow.gateway.constants.MLFLOW_GATEWAY_CLIENT_QUERY_TIMEOUT_SECONDS",
+        new_callable=lambda: 1,
+    ) as mock_timeout:
+        assert mock_timeout == 1
+
+        with mock.patch("requests.Session.request") as mocked_request:
+            mocked_request.side_effect = Timeout
+
+            with pytest.raises(
+                MlflowException, match="The provider has timed out while generating"
+            ):
+                gateway_client.query(route=route, data=data)
+
+            mocked_request.assert_called_once()
```

Comment:
/ai make the following change:

1. Remove assignment to `side_effect`
2. Merge with statements.

Reply:
@mlflow

Thanks for the code!

```
with mock.patch("mlflow.gateway.constants.MLFLOW_GATEWAY_CLIENT_QUERY_TIMEOUT_SECONDS", 1) as mock_timeout, \
    mock.patch("requests.Session.request", side_effect=Timeout) as mocked_request:
    assert mock_timeout == 1
    with pytest.raises(MlflowException, match="The provider has timed out while generating"):
        gateway_client.query(route=route, data=data)

    mocked_request.assert_called_once()
```